src/components/input-switch/array-field.tsx

```diff
@@ -1,108 +1,194 @@
+    }
+  }
+
+
+  const DraggableItemRenderer = (item: any, index: number, disableInput: boolean | undefined) => {
+
+    return <Draggable key={item.id} draggableId={name + '-' + item.id.toString()} index={index} isDragDisabled={disableInput}>
+      {
+        (provided: DraggableProvided) => {
+          return <li className={`item ${getInputType({ name: baseArrayType })} ${!disableInput ? 'add-padding-bottom' : ''}`} ref={provided.innerRef} {...provided.draggableProps} {...provided.dragHandleProps} key={item.id}>
+
+            <div className='move-icon'>
+              <i className='fa-solid fa-grip-vertical'></i>
+            </div>
+
 
-  const placeholder = props.placeholder ? props.placeholder : arrayFieldPlaceholder(props.baseArrayType);
+            <div className='item-input'>
+              <InputSwitch
+                {...props}
+                type={getInputType({ name: baseArrayType })}
+                key={item.id}
+                {...register(`${name}.${index}.val`, {
+                  value: item.val,
+                  onChange: () => trigger(`${name}.${index}.val`)
+                })}
+                displayExtraDateTimeButtons={true}
+                displayDateTimeLabels={baseArrayType === 'date' ? false : true}
+                requiredInput={true}
+                isArrayElement={true}
+              />
+            </div>
+
+            <div>
+              {
+                !disableInput &&
+                <div className='action-buttons'>
+                  <button
+                    type='button' className='fa-solid fa-trash chaise-btn chaise-btn-tertiary chaise-btn-sm'
+                    onClick={() => { remove(index) }}
+                  />
+                </div>
+              }
+            </div>
+          </li>
+        }
+      }
+    </Draggable>
+
+  }
 
   return (
-    <InputField {...props}
-      controllerRules={{
-        validate: arrayFieldValidation
-      }}
-    >
-      {(field, onChange, showClear, clearInput) => (
-        <div className='input-switch-array'>
-          <div className={`chaise-input-control has-feedback ${props.classes} ${props.disableInput ? ' input-disabled' : ''}`}>
-            <textarea
-              placeholder={placeholder} rows={5} className={`${props.inputClasses} input-switch ${
-                hasVerticalScrollbar(textAreaRef.current) ? 'has-scrollbar' : ''
-              }`}
-              {...field} onChange={onChange} disabled={props.disableInput}
-              ref={textAreaRef}
-            />
-            <ClearInputBtn
-              btnClassName={`${props.clearClasses} input-switch-clear ${
-                hasVerticalScrollbar(textAreaRef.current) ? 'has-scrollbar-clear' : ''
-              }`}
-              clickCallback={clearInput}
-              show={showClear && !props.disableInput}
+    <>
+      <div className={`array-input-field-container ${name} ${getInputType({ name: baseArrayType })}`}>
```

Comment:
Adding just the name or the type as class is bad. You should use a more specific name. Something like

```ts
const containerClassName = ['array-input-field-container'];
usedClassNames.push(`${containerClassName[0]}-${name}`);
usedClassNames.push(`${containerClassName[0]}-${getInputType({ name: baseArrayType })}`);

```

So then it would result in something like
```
array-input-field-container array-input-field-container-1-col array-input-field-container-text
```